src/stat.h

```diff
@@ -244,9 +247,15 @@ class CStatHistory : public CStat<DataType, RecordType>
             Start();
     }
 
-    virtual ~CStatHistory() { Stop(); }
+    virtual ~CStatHistory()
+    {
+        op |= STAT_DELETED;
```

Comment:
shouldn't we lock here also?

Reply:
It doesn't matter because op is only read after construction (except for this change during deletion) so there will be no dual read-modify-write conflicts.  And in terms of thread concurrency, op just needs to be set to STAT_DELETED before the lock in Stop() to ensure that it properly interacts with the lock in timeout()